dvc/command/metrics.py

```diff
@@ -12,7 +12,32 @@ def _show(self, metrics):
                 logger.info("{}:".format(branch))
 
             for fname, metric in val.items():
-                logger.info("\t{}: {}".format(fname, metric))
+                self._output_formated_metric(fname, metric)
+
+    def _output_formated_metric(self, fname, metric):
+        content, typ = metric
+
+        delimiter = {"csv": ",", "hcsv": ",", "tsv": "\t", "htsv": "\t"}.get(
+            typ
+        )
+
+        if not delimiter or "\n" not in content:
+            logger.info("\t{}: {}".format(fname, content))
+            return
+
+        logger.info("\t{}:".format(fname))
+
+        rows = [line.split(delimiter) for line in content.split("\n")]
```

Comment:
this won't work for CSV as far as I remember. CSV entry may contain delimiters, newlines, etc. The only proper way to split CSV is to use CSV reader.

Reply:
@shcheklein, csv wasn't working that well, I tried `csv.reader([content], delimiter=builtin_str(delimiter))` and was asking something about universal newline.

I assumed that a simple case like some metrics separated by a comma wouldn't break this so often.

Reply:
I'm not sure I understood why wasn't it working well, could you elaborate a little bit, please?

still, I think if support CSV we must parse it properly, otherwise you can unexpected results and bugs.

For example, will zip(*rows) work if rows have different size (and this will happen if there entries that contain , or \n).

Reply:
```python
reader = csv.reader([content], delimiter=delimiter)

TypeError: '"delimiter" must be string, not unicode'
```

It doesn't support unicode as a delimiter, so we need to use `builtin_str`

```python
content = u'value_mse\tdeviation_mse\tdata_set\n0.421601\t0.173461\ttrain\n0.67528\t0.289545\ttesting\n0.671502\t0.297848\tvalidation'

reader = csv.reader([content], delimiter=builtin_str(delimiter))
rows = [row for row in reader]

*** Error: new-line character seen in unquoted field - do you need to open the file in universal-newline mode?
```

It expects that the content is already _split_ by new lines.

But indeed, it is more reliable because of quoting :+1: .

Reply:
if you want to use CSV reader to read TSV most likely you need to setup it properly - disable, escaping rules, quoting of fields, etc. I would go with simple split for TSV and a reader for CSV.

Reply:
@shcheklein , the current implementation with the `csv.Sniffer` has kind of a "magic setup" (it detects the properties of each line to determine the delimiter and stuff), so I think we could use CSV reader with both TSV and CSV.